docs/builtins-reference.md

```diff
@@ -143,12 +139,13 @@ kmeans(x, centers, iter.max, nstart)
 | String         |The output matrix with the centroids |
 
 
+
+**DML-bodied built-in functions** are written as DML-Scripts and executed as such when called.
 ### Example
 
-KMeans = rand(Matrix x, centers k=0,int iter.max=25,int nstart=10)
-return(Matrix[numerical] C,Matrix[numerical] Y)
+```h2o.kmeans(x = predictors, k = 100, estimate_k = T, standardize = F,
```

Comment:
Are you able to run this code in the colab notebook provided to you, if yes screenshot please.

Also, this seems to be a copy-paste from `H2O` software, different from `SystemDS`.

Reply:
I'm unable to find a small code snippet for Kmeans function in R which runs without any error on Collab. I tried with a lot of examples.

Reply:
We need to closely understand the this function:
https://github.com/apache/systemml/blob/faec15810e18fc497e950b830811a919f5aa5c5f/scripts/builtin/kmeans.dml#L46

Think about the inputs required and initialize the required arguments.
We can take already implemented functions as examples. 
This task is tough and iterative, copy-paste solutions are not available. Some people spend whole days to find the correct combinations, though. 🙂

Reply:
Alright . I'm on it.

Reply:
The syntax errors are quite persistent . No example worked in the collab till now.

Reply:
```
X = rand (rows = 10, cols = 10)
kmeans(X = X, k = 0, runs = 10, max_iter = 1000, )
```

Reply:
![Screenshot (144)](https://user-images.githubusercontent.com/54388379/83662574-4e6bc180-a5e5-11ea-851d-8df36024a06f.png)

Reply:
The snippet, I've provided above is a DML code to run. 🙂

Reply:
Kindly have a look at how other functions are documented.